### ISSUE TITLE
mysql feature proposal

### ISSUE THREAD

**Issue by georgeartem** [dogecoinfoundation/gigawallet#115]
Have some concerns about management and administration of sqlite on an enterprise level, particularly as it relates to user authorization. Authentication appears to be relatively easy to set up, but there do not appear to be any permissions features - some might consider this an issue. 

Additionally, the db is based on a single file, and although backup and restore features appear to be available, I think naturally, a product owner might have some concerns about security and redundancy.

Would it be useful to create a module that supports connection with mysql as an alternative to sqlite implementation? I'm relatively confident that this is something I could knock out in .go with a little guidance.

**Comment by qlpqlp** [comment]
It already enables you to run on postgreSQL and like on MySQL, you can set permissions :)

![image](https://github.com/dogecoinfoundation/gigawallet/assets/5968076/456ff631-fe7c-4f28-ae38-5ade3396b4b2)

https://gigawallet.dogecoin.org/installation.html

For backups of the DB, using postgreSQL  (that I recommend using it), on Ubuntu, that I recommended a few minutes ago on your other post to run on Windows Subsystem, you can install the pgagent to automate backups :)

Here it is a wonderful tutorial how to
https://www.digitalocean.com/community/tutorials/how-to-schedule-automatic-backups-for-postgresql-with-pgagent-in-pgadmin

Hope this helps :)

**Comment by georgeartem** [comment]
Thanks Paulo, I am glad you are supporting a more robust db. It would be really great if these external docs were in the repo.

Nevertheless you are going to invariably find dbadmin types who love MySQL and are going to be unwilling and/or upset about having to use a different SQL flavor. 

I think I may be better off learning how to use and compile in docker rather than the windows subsystem.

Hopefully future releases can support Windows out of the box.

What do you think?

**Comment by qlpqlp** [comment]
Give me a couple of hours and I will try to make a Docker image :)

About MySQL, I always used MySQL since ~2000 I think I use it (in the last years MariaDB lol) so I know what your are saying :D

**Comment by georgeartem** [comment]
Cool. I think I saw you running something on X that looked like it could have been Docker. Success!

**Comment by qlpqlp** [comment]
yes, I already can run successfull GigaWallet but for now only with Sqlite 😅

Have a new version now that already installed successfully with postgreSQL but have some issues on local network connection that I have to try to figure it out in the morning to then share the Docker with both options 😁

**Comment by georgeartem** [comment]
Nice, the point I was trying to make about Windows is that there are, I don’t know what percentage, but a lot of Shibes who are running nodes in that environment and might like to learn how to use GigaWallet. 

A cross environment install would be an unnecessary added hurdle particularly with how long it takes for that application to load up the chain.
